html/src/THtml.cxx

```diff
@@ -1746,8 +1746,8 @@ void THtml::CreateListOfClasses(const char* filter)
       if (!module) {
          bool moduleSelected = cdi->IsSelected();
 
-         TString parentModuleName(gSystem->DirName(modulename));
-         TModuleDocInfo* super = 0;
+         TString parentModuleName = gSystem->GetDirName(modulename);
```

Comment:
```suggestion
         TString parentModuleName(gSystem->GetDirName(modulename));
```

Reply:
I'm not a big fan of such a change. The assembly is strictly identical and it reads worse (function call?). If assignment (my preferred) is really that terrible please at least consider using braced-init `TString parentModuleName{gSystem->GetDirName(modulename)};

Reply:
Yes, I also had same in mind when replacing `TString res(value);` by `TString res = value;` Should we keep as is?

Reply:
> and it reads worse (function call?)

Yes.

> The assembly is strictly identical see link

How can it be? TString::operator= is outlined and thus must be called.

Reply:
but isn't ... because the operator= only applies to existing object not during construction .. so never mind :)